ui/app/AppLayouts/Wallet/services/dapps/types/SessionRequest.qml

```diff
@@ -62,6 +64,86 @@ QtObject {
         readonly property var all: [personalSign, sign, signTypedData_v4, signTypedData, sendTransaction]
     }
 
+    enum ErrorCode {
+        NoError,
+        InvalidAccount,
+        InvalidChainId,
+        InvalidData,
+        InvalidMessage,
+        InvalidMethod,
+        UnsupportedMethod,
+        InvalidRequest,
+        RuntimeError,
+        Ignored
+    }
+    /* 
+    Function parsing the session request event
+    Throws exception if the event is invalid
+    returns { request: [Object], error: [ErrorCode] }
+        request: {
+            event,
+            topic,
+            requestId,
+            method,
+            account,
+            chainId,
+            data,
+            preparedData,
+            expiryTimestamp,
+            transaction: {
+                value,
+                maxFeePerGas,
+                maxPriorityFeePerGas,
+                gasPrice,
+                gasLimit,
+                nonce
+            }
+        }
+    */
+    function parse(event, hexToDec = (hex) => { return parseInt(hex, 16) }) {
```

Comment:
Now all the session request parsing happens in this `singletone` component. There's also a storybook page where we can paste any WC/BC session request to investigate the resulted object.